bazel/BUILD

```diff
@@ -1,3 +1,8 @@
 package(default_visibility = ["//visibility:public"])
 
 exports_files(["gen_sh_test_runner.sh"])
+
+config_setting(
+    name = "disable_tcmalloc",
+    values = {"linkopt": "-fsanitize=address"},
```

Comment:
nit: Instead of keying off of this specific option, should we actually set some other option when we are doing sanitizing? Should we also allow tcmalloc to be disabled by a generic option? (For example, newer valgrinds are totally broken with tcmalloc).